connect/api/src/main/java/org/apache/kafka/connect/data/SchemaProjector.java

```diff
@@ -109,16 +109,12 @@ private static Object projectStruct(Schema source, Struct sourceStruct, Schema t
                     Object targetFieldValue = project(sourceField.schema(), sourceFieldValue, targetField.schema());
                     targetStruct.put(fieldName, targetFieldValue);
                 } catch (SchemaProjectorException e) {
-                    throw new SchemaProjectorException("Error projecting " + sourceField.name(), e);
+                    throw new SchemaProjectorException("Error projecting " + fieldName, e);
                 }
-            } else {
-                Object targetDefault;
-                if (targetField.schema().defaultValue() != null) {
-                    targetDefault = targetField.schema().defaultValue();
-                } else {
-                    throw new SchemaProjectorException("Cannot project " + source.schema() + " to " + target.schema());
-                }
-                targetStruct.put(fieldName, targetDefault);
+            } else if (targetField.schema().defaultValue() != null) {
+                targetStruct.put(fieldName, targetField.schema().defaultValue());
+            } else if (!targetField.schema().isOptional()) {
+                throw new SchemaProjectorException("Required field `" +  fieldName + "` is missing from source schema: " + source);
             }
```

Comment:
Is there a good reason why we now leave the code without a "default" `else` branch?